File: week-04/day-02/Aircraft/Aircraft.py
class Aircraft(object):
    ammo = 0
    def __init__(self, aircraft_type):
        self.aircraft_type = aircraft_type
        if aircraft_type == "F16":
            self.maxammo = 8
            self.basedamage = 30
        elif aircraft_type == "F35":
            self.maxammo = 12
            self.basedamage = 50
    

    def fight(self):
        max_damage = self.ammo * self.basedamage
        self.ammo = 0
        return max_damage

    
    def refill(self, number):
        if self.aircraft_type == "F16":
            if self.maxammo > self.ammo:
                number -= (self.maxammo-self.ammo)
                return number
            else:
                return number
        elif self.aircraft_type == "F35":
            if self.maxammo > self.ammo:
                number -= (self.maxammo-self.ammo)
                return number
            else:
                return number


    def getType(self):
        print(self.aircraft_type)

    
    def getStatus(self):
        print("Type {}, Ammo: {}, Base Damage: {}, All Damage: {}".format(self.aircraft_type, self.ammo, self.basedamage, (self.maxammo*self.basedamage)))
    

File: Csharp-course/week-03/day-05/AircraftCarrier/AircraftCarrier/Aircraft.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AircraftCarrier
{
    class Aircraft
    {
        public int maxAmmo;
        public int baseDamage;
        public int currentAmmo;
        public string type;

        public Aircraft(string _type)
        {
            this.type = _type;
            this.currentAmmo = 0;
            if (_type == "F16")
            {
                this.maxAmmo = 8;
                this.baseDamage = 30;
            }
            else if (_type == "F35")
            {
                this.maxAmmo = 12;
                this.baseDamage = 50;
            }
        }

        public int Fight()
        {
            return this.currentAmmo * this.baseDamage;
        }

        public int Refill(int givenAmmo)
        {
            if (this.maxAmmo > this.currentAmmo)
            {
                int usedAmmo = (this.maxAmmo - this.currentAmmo);
                this.currentAmmo += usedAmmo;
                return usedAmmo;
            }
            return 0;
        }

        public string getType()
        {
            return this.type;
        }

        public bool isPriority()
        {
            if (this.type == "F35")
            {
                return true;
            }
            return false;
        }
    }
}
